pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -742,19 +742,28 @@ protected void internalCreatePartitionedTopic(AsyncResponse asyncResponse, int n
      * @param topicName given topic name
      */
     protected CompletableFuture<Boolean> checkTopicExistsAsync(TopicName topicName) {
-        return pulsar().getNamespaceService().getListOfTopics(topicName.getNamespaceObject(),
-                CommandGetTopicsOfNamespace.Mode.ALL)
-                .thenCompose(topics -> {
-                    boolean exists = false;
-                    for (String topic : topics) {
-                        if (topicName.getPartitionedTopicName().equals(
-                                TopicName.get(topic).getPartitionedTopicName())) {
-                            exists = true;
-                            break;
+        if (topicName.isPersistent()) {
+            String path = ZkAdminPaths.partitionedTopicPath(topicName);
+            return pulsar().getPulsarResources().getNamespaceResources().getPartitionedTopicResources().
+                    existsAsync(path).thenCombine(pulsar().getLocalZkCacheService().
+                            managedLedgerExists(topicName.getPersistenceNamingEncoding()),
+                    (isPartitionedExists, isNonPartitionedExists) -> isPartitionedExists || isNonPartitionedExists);
+
+        } else {
+            return pulsar().getNamespaceService().getListOfTopics(topicName.getNamespaceObject(),
+                    CommandGetTopicsOfNamespace.Mode.NON_PERSISTENT)
```

Comment:
We have multiLayerTopicsMap in BrokerService, we don't need to traverse all NON_PERSISTENT, and then traverse again to determine whether it exists.
We only need to traverse the bundles, and then directly map.contains can do it, usually the number of bundles is not too much